app/notification.go

```diff
@@ -102,9 +102,10 @@ func SendNotifications(post *model.Post, team *model.Team, channel *model.Channe
 
 		// find which users in the channel are set up to always receive mobile notifications
 		for _, profile := range profileMap {
-			if profile.NotifyProps[model.PUSH_NOTIFY_PROP] == model.USER_NOTIFY_ALL &&
+			if (profile.NotifyProps[model.PUSH_NOTIFY_PROP] == model.USER_NOTIFY_ALL &&
 				(post.UserId != profile.Id || post.Props["from_webhook"] == "true") &&
-				!post.IsSystemMessage() {
+				!post.IsSystemMessage()) ||
```

Comment:
I think the `(post.UserId != profile.Id || post.Props["from_webhook"] == "true") && !post.IsSystemMessage()` Shouldn't be inside the brackets. Ex you never want to recieve pushes for system messages.